src/ray/gcs/redis_module/ray_redis_module.cc

```diff
@@ -407,6 +412,109 @@ int ChainTableAppend_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv,
 }
 #endif
 
+int Set_DoPublish(RedisModuleCtx *ctx, RedisModuleString **argv, bool is_add) {
+  RedisModuleString *pubsub_channel_str = argv[2];
+  RedisModuleString *id = argv[3];
+  RedisModuleString *data = argv[4];
+  // Publish a message on the requested pubsub channel if necessary.
+  TablePubsub pubsub_channel;
+  REPLY_AND_RETURN_IF_NOT_OK(ParseTablePubsub(&pubsub_channel, pubsub_channel_str));
+  if (pubsub_channel != TablePubsub::NO_PUBLISH) {
+    // All other pubsub channels write the data back directly onto the
+    // channel.
+    return PublishTableUpdate(ctx, pubsub_channel_str, id,
+                              is_add ? GcsTableNotificationMode::APPEND_OR_ADD
+                                     : GcsTableNotificationMode::REMOVE,
+                              data);
+  } else {
+    return RedisModule_ReplyWithSimpleString(ctx, "OK");
+  }
+}
+
+int Set_DoWrite(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, bool is_add,
+                bool *changed) {
+  if (argc != 5) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  RedisModuleString *prefix_str = argv[1];
+  RedisModuleString *id = argv[3];
+  RedisModuleString *data = argv[4];
+
+  RedisModuleString *key_string = PrefixedKeyString(ctx, prefix_str, id);
+  RedisModuleCallReply *reply =
+      RedisModule_Call(ctx, is_add ? "SADD" : "SREM", "ss", key_string, data);
```

Comment:
`RedisModule_Call` seems slower than the direct c++ interface. Is that because there is no such interface ready for `SREM`? If that is the case, please add a TODO comment to replace this command to C++ interface if hredis is updated..

Reply:
Yes, set type API is not available yet. See [https://redis.io/topics/modules-intro](url)